### PR TITLE
skip migration tests if no migration is required

### PR DESCRIPTION
**Pull request by icewind1991**
cc @DeepDiver1975 @PVince81

### CODE REVIEW

File: lib/private/db/migrator.php [hunk 1]
```diff
@@ -254,4 +258,19 @@ protected function convertStatementToScript($statement) {
 	protected function getFilterExpression() {
 		return '/^' . preg_quote($this->config->getSystemValue('dbtableprefix', 'oc_')) . '/';
 	}
+
+	public function needsMigration(Schema $targetSchema) {
+		$diff = $this->getDiff($targetSchema, $this->connection);
```
Comment (PVince81) [inline lib/private/db/migrator.php]: so basically we're doing the table diff twice ?
I assume this won't cause any performance issue

**Comment by PVince81** [comment]
Thanks, code makes sense. Will test later.

**Comment by PVince81** [comment]
Please note that this might not work yet, I discovered that the schema diff is with the new Doctrine is detecting index changes, even if there aren't, see https://github.com/owncloud/core/issues/19317

**Comment by PVince81** [comment]
@icewind1991 hmm, your fix will still make a copy of all the tables even if only one has changed.
You need to add a check inside "checkTableMigrate()", before doing the copy.

**Comment by icewind1991** [comment]
Changed the upgrade check logic to only check changed tables

**Comment by icewind1991** [comment]
(I think the failing pgsql test is related to the index issue @PVince81 is working on)

**Comment by PVince81** [comment]
- [ ] blocked by https://github.com/owncloud/core/pull/19347

**Comment by rullzer** [comment]
Moving to 9.0 since #19347 is 9.0 now

**Comment by rullzer** [comment]
@cmonteroluque to 9.0?

**Comment by cmonteroluque** [comment]
@rulizer yeah

**Comment by DeepDiver1975** [comment]
@icewind1991 where are we going with this? conflicting and moved around since 8.2

**Comment by PVince81** [comment]
Requires https://github.com/owncloud/core/pull/19347

**Comment by DeepDiver1975** [comment]
If we go for doctrine migrations all this schema copy and testing is no longer required.

**Comment by DeepDiver1975** [comment]
setting 9.2 to get this sorted out

**Pull request closed without merging**